src/Akeneo/Pim/Enrichment/Product/back/Test/Specification/Infrastructure/Validation/UserIntentsShouldBeUniqueValidatorSpec.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Specification\Akeneo\Pim\Enrichment\Product\Infrastructure\Validation;
+
+use Akeneo\Pim\Enrichment\Product\API\Command\UserIntent\SetTextValue;
+use Akeneo\Pim\Enrichment\Product\Infrastructure\Validation\UserIntentsShouldBeUnique;
+use Akeneo\Pim\Enrichment\Product\Infrastructure\Validation\UserIntentsShouldBeUniqueValidator;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Symfony\Component\Validator\Constraints\Type;
+use Symfony\Component\Validator\ConstraintValidatorInterface;
+use Symfony\Component\Validator\Context\ExecutionContext;
+use Symfony\Component\Validator\Violation\ConstraintViolationBuilderInterface;
+
+class UserIntentsShouldBeUniqueValidatorSpec extends ObjectBehavior
+{
+    function let(ExecutionContext $context)
+    {
+        $this->initialize($context);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(UserIntentsShouldBeUniqueValidator::class);
+        $this->shouldImplement(ConstraintValidatorInterface::class);
+    }
+
+    function it_throws_an_exception_with_a_wrong_constraint()
+    {
+        $this->shouldThrow(\InvalidArgumentException::class)->during('validate', [1, new Type([])]);
+    }
+
+    function it_does_nothing_when_the_value_intents_are_distinct(ExecutionContext $context)
+    {
+        $context->addViolation(Argument::cetera())->shouldNotBeCalled();
```

Comment:
addViolation is not directly called. We should test buildViolation instead

```suggestion
        $context->buildViolation(Argument::cetera())->shouldNotBeCalled();
```